File: pinyougou_2/pinyougou_parent/pinyougou_cart_interface/src/main/java/com/pinyougou/cart/service/CartService.java
package com.pinyougou.cart.service;

import com.pinyougou.vo.Cart;

import java.util.List;

public interface CartService {


    public List<Cart> addGoodsToCartList(List<Cart> cartList,Long itemId,Integer num);

    public List<Cart> findFromRedis(String username);

    public void addGoodsToRedis(String username, List<Cart> cartList);

    public List<Cart> mergeCartList(List<Cart> cartList1, List<Cart> cartList2);

}

File: pinyougou_2/pinyougou_parent/pinyougou_cart_web/src/main/webapp/js/service/cartService.js
//服务层
app.service('cartService',function($http){
     this.findCartList=function () {
         return  $http.get("../cart/findCartList.do")
     }

    this.addGoodsToCartList = function (itemId, num) {
        return $http.get("../cart/addGoodsToCartList.do?itemId="+itemId+"&num="+num)
    };


    this.countTotal=function (cartList) {
        var totalValue = {totalCount: 0, totalMoney: 0};
        for(var i=0;i<cartList.length;i++) {
            var cart=cartList[i]
            for(var j=0;j<cart.orderItemList.length;j++) {
                totalValue.totalCount+=cart.orderItemList[j].num;
                totalValue.totalMoney+=cart.orderItemList[j].totalFee;
            }
        }

        return totalValue;
    }

    //通过用户名查询地址列表,用户名由security提供
    this.findAddressList=function () {
        return $http.get("../address/findAddressByUsername.do")
    }
});